conformance/lang/expressions/trap_expr.balt

```diff
@@ -0,0 +1,252 @@
+    io:println(t); // @output error("msg")
+}
+
+Test-Case: output
+Description: Test trap-expr when the expression is a member-access-expr.
+Labels: trap-expr, error-type, int, array-type, member-access-expr
+
+function init() {
+    int[] i = [];
+    io:println(trap i[3]); // @output error("{ballerina/lang.array}IndexOutOfRange",message="array index out of range: index: 3, size: 0")
+}
+
+Test-Case: output
+Description: Test trap-expr when the expression is a function-call-expr.
+Labels: trap-expr, error-type, int, function-call-expr, array-type, union-type, error-constructor-expr
+
+function foo() returns int {
+    int[] i = [];
+    return i[3];
+}
+
+function bar() {
+    panic error("msg");
+}
+
+function init() {
+    int|error t1 = trap foo();
+    io:println(t1); // @output error("{ballerina/lang.array}IndexOutOfRange",message="array index out of range: index: 3, size: 0")
+    error? t2 = trap bar();
+    io:println(t2); // @output error("msg")
+}
+
+Test-Case: output
+Description: Test trap-expr when the expression is a type-cast-expr.
+Labels: trap-expr, error-type, int, string, function-call-expr, union-type, type-cast-expr
+
+function foo() returns int|string {
+    return "str";
+}
+
+function init() {
+    int|error t1 = trap <int>foo();
+    io:println(t1); // @output error("{ballerina}TypeCastError",message="incompatible types: 'string' cannot be cast to 'int'")
+}
+
+Test-Case: output
+Description: Test trap-expr when the expression is a unary-logical-expr.
+Labels: trap-expr, error-type, int, boolean, function-call-expr, union-type
+
+function foo() returns int|boolean {
+    return 23;
+}
+
+function init() {
+    boolean|error t1 = trap !<boolean>foo();
+    io:println(t1); // @output error("{ballerina}TypeCastError",message="incompatible types: 'int' cannot be cast to 'boolean'")
+}
+
+Test-Case: output
+Description: Test trap-expr when the expression is a multiplicative-expr.
+Labels: trap-expr, error-type, int, boolean, function-call-expr, multiplicative-expr
+
+function foo() returns int {
+    return 0;
+}
+
+function init() {
+    int|error t1 = trap 2 / foo();
+    io:println(t1); // @output error("{ballerina}DivisionByZero",message=" / by zero")
+}
+
+Test-Case: output
+Description: Test trap-expr when the expression is a trap-expr.
+Labels: trap-expr, error-type, int, boolean, function-call-expr, multiplicative-expr
+
+function foo() returns int {
+    return 0;
+}
+
+function init() {
+    int|error t1 = trap trap 2 / foo();
+    io:println(t1); // @output error("{ballerina}DivisionByZero",message=" / by zero")
+}
+
+Test-Case: error
+Description: Check the type of trap-expr.
+Labels: trap-expr, error-type, int, function-call-expr, multiplicative-expr
+
+function foo() returns int {
+    return 0;
+}
+
+function errorFunction() {
+    int _ = trap 2 / foo(); // @error incompatible types: expected 'int', found '(int|error)'
+    error e = trap 2 / foo(); // @error incompatible types: expected 'error', found 'int'
```

Comment:
Isn't the test overly complicated? We don't need a function call. Even a variable reference would suffice.

Reply:
Changed